native/src/main/java/io/ballerina/lib/sap/BallerinaDestinationDataProvider.java

```diff
@@ -0,0 +1,298 @@
+package io.ballerina.lib.sap;
+
+import com.sap.conn.jco.ext.DestinationDataEventListener;
+import com.sap.conn.jco.ext.DestinationDataProvider;
+import io.ballerina.runtime.api.utils.StringUtils;
+import io.ballerina.runtime.api.values.BMap;
+import io.ballerina.runtime.api.values.BString;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+public class BallerinaDestinationDataProvider implements DestinationDataProvider {
+
+    private final Map<String, Properties> destinationProperties = new HashMap<>();
+
+    @Override
+    public Properties getDestinationProperties(String destinationName) {
+        if (destinationProperties.containsKey(destinationName)) {
+            return destinationProperties.get(destinationName);
+        } else {
+            throw new RuntimeException("Destination " + destinationName + " not found");
+        }
+    }
+
+    @Override
+    public void setDestinationDataEventListener(DestinationDataEventListener eventListener) {
+    }
+
+    @Override
+    public boolean supportsEvents() {
+        return true;
+    }
+
+    public void addDestination(BMap<BString, Object> jcoDestinationConfig) {
+        Properties properties = new Properties();
+        properties.setProperty(DestinationDataProvider.JCO_CLIENT, jcoDestinationConfig.getStringValue(
+                StringUtils.fromString("jcoClient")).toString());
+        properties.setProperty(DestinationDataProvider.JCO_USER, jcoDestinationConfig.getStringValue(
+                StringUtils.fromString("user")).toString());
+        properties.setProperty(DestinationDataProvider.JCO_PASSWD, jcoDestinationConfig.getStringValue(
+                StringUtils.fromString("password")).toString());
+        properties.setProperty(DestinationDataProvider.JCO_LANG, jcoDestinationConfig.getStringValue(
+                StringUtils.fromString("language")).toString());
+        properties.setProperty(DestinationDataProvider.JCO_ASHOST, jcoDestinationConfig.getStringValue(
+                StringUtils.fromString("host")).toString());
+        properties.setProperty(DestinationDataProvider.JCO_SYSNR, jcoDestinationConfig.getStringValue(
+                StringUtils.fromString("systemNumber")).toString());
+        properties.setProperty(DestinationDataProvider.JCO_GROUP, jcoDestinationConfig.getStringValue(
```

Comment:
Lets use constants for this

Reply:
Changed with [1587866](https://github.com/ballerina-platform/module-ballerinax-sap.jco/pull/2/commits/1587866408961566b051f449be7fd7303ce85a9f).